concepts/default-enarx-trust-process.md

```diff
@@ -0,0 +1,480 @@
+explicitly uni-directional (though this does not preclude a
+corresponding trust relationship in the other direction).  In the case
+of Enarx, it is trust relationships from the workload entity to other
+entities (primarily the host owner in this document), and maintenance
+of the tenant's trust domain, which are always the priority - any
+other benefits which may accrue to other parties, are considered
+side benefits.
+
+On a different note, this RFC deals mainly with the communications
+between the various Enarx components.  Although the Orchestrator is
+involved and mentioned in this document, it is not the document's focus.
+
+## Trust state transitions in Enarx
+
+See rfc#0000x for an introduction to trust relationships.
+See rfc#0000x for an introduction to trust domains and their application
+to Enarx.
+See rfc#0000x for an introduction to endorsing authorities, trust anchors
+and trust pivots.
+
+rfc#0000x described the following Enarx use case states for the default
+Enarx trust model (TODO: check correctness in rfc#0000x and normalise):
+ - State 0 - initial state
+ - State 1 - pre-attestation
+ - State 2 - attested
+ - State 3 - provisioned
+
+Non-Enarx use cases were also discussed, but these are irrelevant to this
+discussion, and therefore ignored in this RFC.
+
+The changes between states were characterised as the transition of
+particular components between trust domains:
+ 1. host owner trust domain;
+ 2. tenant trust domain.
+
+This RFC uses the same states, trust domains and components.  Although
+the level of description may feel overly exhaustive, the members of the
+Enarx project believe that it is important to be explicit about the trust
+models being implemented, and how they reflect the "real world", in order
+to allow designers, coders, documenters, testers, auditors and users of
+the project to have as full an understanding of the process and
+architecture as possible.
+
+### How the trust domain change happens
+The key state change in terms of the Enarx default trust model is from
+state 1 to state 2.  It is at this point that at **TEE instance** on the
+host (and in the **host owner's trust domain**), loaded with an **Enarx
+runtime image** (which then executes, to become an **Enarx runtime
+instance**) becomes an **Empty Keep**, within the **tenant trust domain**.
+This is achieved by the use of a __trust pivot__: the **host CPU + CPU
+firmware**.  The attestation process described in the State 1->2 change
+section explicitly notes that that cryptographic identity of the **host
+CPU + CPU firmware** must be validated as part of the attestation
+measurement check process.
+
+The ability for the tenant to trust the host CPU + CPU firmware underpins
+the entire trust model process, and it is vital that all implementations
+check it very carefully.
+
+### State 0->1 change
+
+#### State 0 - initial state
+
+ - host owner trust domain
+    - host
+    - host CPU + CPU firmware
```

Comment:
Seems a little redundant. When will we have one of these without the other?

Reply:
We won't (I suspect!), but I'm trying to be as explicit at possible.

Reply:
And, more important, we're utilising a combination of the capabilities that the two entities provide.